client/api.c

```diff
@@ -920,6 +921,25 @@ TDNFAddCmdLinePackages(
     goto cleanup;
 }
 
+static void
```

Comment:
This function prepends a `*` always? So "foo/bar" becomes "*foo/bar"?  Why?

Reply:
I understand what you are trying to achieve here, but I think this needs more work. For example, you can use the `PATH` for executables. I think `dnf` already does something like this - find out how and maybe we can emulate that.

Reply:
I any case, add comments about what this is doing for future generations.

Reply:
dnf5 handles this in a slight complex way, https://github.com/rpm-software-management/dnf5/blob/main/dnf5/commands/provides/provides.cpp#L78

I think prepending `*` is easier and more user friendly way to achieve the same result. And users can simply do `tdnf whatprovides/provides grub` (for instance) and they get desired results. I don't like doing things on behalf of users, that's a recipe for failure at some point but I think this is one exemption. What say you?

I also improved the code slightly, if user has provided `*` as first character in the query, the string doesn't get modified.

Reply:
I don't like this. I think this can be really annoying when `tdnf` prints out tons of matches just because it always prepends a wildcard. I think we should do it the way dnf does it. We could even improve on it by using the `PATH` env var, and maybe do something similar for libraries.